packages/text-editor/src/react/source-extension-react.tsx

```diff
@@ -81,6 +97,10 @@ export const SourceExtensionReact = SourceExtension.extend<
 		this.storage.nodes = this.options.nodes;
 	},
 
+	onCreate() {
+		this.storage.nodes = this.options.nodes;
+	},
+
```

Comment:
:memo: Added onCreate to ensure storage is properly initialized after cleanup, enabling correct node reference insertion via @ button.